### ISSUE TITLE
Using surround to manipulate quotes is broken when emacs doesn't think you're in a quoted string

### ISSUE THREAD

**Issue by rolandcrosby** [timcharper/evil-surround#56]
In this text file, with your cursor in the middle of the word 'this':

```
5'11" tall
change "this" word
```

attempting to manipulate the surrounding quotes (`ds"`, `cs"`, etc) returns a "No quoted string found" error. In Vim with surround installed, the quotes are correctly detected and manipulated.

I believe this is due to Evil's quote textobject using native Emacs quote parsing, per [this thread](https://bitbucket.org/lyro/evil/issue/504/weird-behaviour-of-evil-inner-single-quote). Also per that thread, it sounds like Evil itself is not going to change the way quote textobjects are parsed, so a change to make surround behave normally would have to be made in evil-surround itself.

**Comment by ninrod** [comment]
Yes this is related to #92. I think evil-surround should enclose it's scope only inside a line, not multlinewise because it get's hard to know if you are writing a multiline string or not.

**Comment by ninrod** [comment]
@rolandcrosby, this is really hard for evil-surround to deal with. vim surround actually displays disparate behaviour in this regard. Let's take this as an example(cursor on `|`):

```text
5'11( tall
ch|ange )this" word
```

`ds(` will kill the parenthesis in vim-surround. Now with quotes, it will not. I'm not sure if vim-surround is right. I think evil-surround is right in this case because it treats all cases equally: multilinewise.

I'm not a 100% sure of what to do in this case but I'm inclined to leave it as it is.

**Comment by ninrod** [comment]
what do you think @justbur, @edkolev, @wasamasa and @TheBB?

**Comment by ninrod** [comment]
@actionshrimp I think that happens because yaml-mode defines a different [syntax table](https://github.com/yoshiki/yaml-mode/blob/master/yaml-mode.el#L247-L279) for quoted stuff.

**Comment by actionshrimp** [comment]
Ah I just realised this is related to evil text-objects rather than surround specifically (I was actually trying to do some surround related operations which sent me off in the wrong place). Sorry for the noise!

**Comment by jvtrigueros** [comment]
@actionshrimp I just hit this exact issue in YAML, did you have a workaround for this? Don't mean to hijack, just working in YAML so much and I don't know where else to look.
